apps/api/src/routers/saml.py

```diff
@@ -81,6 +83,13 @@ async def _get_saml_auth(req: Request) -> OneLogin_Saml2_Auth:
     return OneLogin_Saml2_Auth(request_data, old_settings=settings)
 
 
+async def _insert_native_record(user: NativeUser) -> None:
+    now = utc_now()
+    await mongodb_handler.update_one(
+        Collection.USERS, {"_id": user.uid}, {"last_login": now}, upsert=True
+    )
```

Comment:
Thought [non-blocking]: I just discovered MongoDB supports [`$currentDate`](https://www.mongodb.com/docs/manual/reference/operator/update/currentDate/) as an update operator. I think for consistency with `guest_auth._remove_guest_key`, we can keep this for now.